### PR TITLE
Notification em notif table crud

### PR DESCRIPTION
**Pull request by AleeWeeR**
Review requested for em_notif notification seat equipment tables

### CODE REVIEW

File: src/main/java/com/exadel/sandbox/em_notif/service/impl/EmployeeNotificationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        List<EmployeeNotification> employeeNotificationList = emplyeeNotificationRepository.findAll();
+
+        List<EmployeeNotificationResponseDto> employeeNotificationResponseDtos = new ArrayList<>();
+
+        for (EmployeeNotification emplNotif:
+             employeeNotificationList) {
+            employeeNotificationResponseDtos.add(fullMap(emplNotif));
+        }
+
+        return employeeNotificationResponseDtos;
```
Comment (KirvelPasha) [inline src/main/java/com/exadel/sandbox/em_notif/service/impl/EmployeeNotificationServiceImpl.java]: Please rewrite this method something like that:
 List<EmployeeNotification> employeeNotificationList = emplyeeNotificationRepository.findAll();

        return employeeNotificationList.stream()
                .map(this::fullMap)
                .collect(Collectors.toList());

And also please check other places and try to use stream where possible

File: src/main/java/com/exadel/sandbox/em_notif/service/impl/EmployeeNotificationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    public EmployeeNotificationResponseDto getEmNotifById(Long id) {
+
+        Optional<EmployeeNotification> byId = emplyeeNotificationRepository.findById(id);
+
+        EmployeeNotification employeeNotification = byId.orElseThrow( ()-> new EntityNotFoundException("employee_notification with id: " + id + " not found"));
+
+
+        return fullMap(employeeNotification);
+
+
```
Comment (KirvelPasha) [inline src/main/java/com/exadel/sandbox/em_notif/service/impl/EmployeeNotificationServiceImpl.java]: Please apply code formatting for all classes

File: src/main/java/com/exadel/sandbox/notification/service/impl/NotificationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        return fullMap(notificationRepository.save(notification));
+    }
+
+
+
+    private NotificationResponseDto fullMap(Notification notification) {
+        NotificationResponseDto notificationResponseDto = mapper.map(notification, NotificationResponseDto.class);
+
+
+        return null;
```
Comment (KirvelPasha) [inline src/main/java/com/exadel/sandbox/notification/service/impl/NotificationServiceImpl.java]: Please check this method.

File: src/main/java/com/exadel/sandbox/em_notif/repository/EmplyeeNotificationRepository.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.exadel.sandbox.em_notif.repository;
+
+import com.exadel.sandbox.em_notif.entity.EmployeeNotification;
+import org.springframework.data.jpa.repository.JpaRepository;
+
```
Comment (KirvelPasha) [inline src/main/java/com/exadel/sandbox/em_notif/repository/EmplyeeNotificationRepository.java:6]: Please add annotation

File: build.gradle [hunk 5]
```diff
@@ -37,6 +37,9 @@ dependencies {
     api('org.springframework.boot:spring-boot-starter-web')
     api('org.springframework.boot:spring-boot-starter-security')
     implementation('org.mariadb.jdbc:mariadb-java-client')
+    // https://mvnrepository.com/artifact/mysql/mysql-connector-java
+    implementation group: 'mysql', name: 'mysql-connector-java', version: '8.0.28'
```
Comment (ikantsevich) [inline build.gradle]: please do not use another connector

File: src/main/java/com/exadel/sandbox/em_notif/contoller/EmployeeNotificationController.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @GetMapping("{id}")
+    EmployeeNotificationResponseDto getById(@PathVariable("id") Long id) {
+
+        return employeeNotificationService.getEmNotifById(id);
+    }
+
+    @PostMapping()
+    EmployeeNotificationResponseDto createEmployeeNotification(@RequestBody EmployeeNotificationBaseDto employeeNotificationBaseDto) {
```
Comment (ikantsevich) [inline src/main/java/com/exadel/sandbox/em_notif/contoller/EmployeeNotificationController.java:31]: U need 4 DTOs
1. ResponseDTO (with not updating fields)
2. CreateDTO with fields for create
3. Update DTO with fields for update
4.BaseDTO to take out common fields

File: src/main/java/com/exadel/sandbox/em_notif/dto/EmployeeNotificationBaseDto.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import lombok.NoArgsConstructor;
+
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+//@EqualsAndHashCode(callSuper = true)
+public class EmployeeNotificationBaseDto {
+
+    //FIXME employee id should be from employee dto
```
Comment (ikantsevich) [inline src/main/java/com/exadel/sandbox/em_notif/dto/EmployeeNotificationBaseDto.java]: What is it?

**Review by ikantsevich — DISMISSED** [review]

File: src/main/resources/application.yml [hunk 8]
```diff
@@ -2,11 +2,11 @@ server:
   port: 8081
   servlet.context-path: /sandbox
 datasource:
-  schema: 'sandbox'
-  host: '${dockerdbhost:localhost}:3306'
+  schema: 'workplace'
+  host: '${dockerdbhos:localhost}:3306'
```
Comment (ghost) [inline src/main/resources/application.yml]: Please set back the configuration

File: src/main/resources/db.changelog/changes/init.yaml [hunk 9]
```diff
@@ -2,9 +2,28 @@ databaseChangeLog:
   - logicalFilePath: no-path
   - changeSet:
       id: init change set
-      author: ikantsevich@exadel.com
+      author: Alisher
```
Comment (ghost) [inline src/main/resources/db.changelog/changes/init.yaml]: Alisher I think you can leave this init file and create new one for yourself

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by believer-0910** [comment]
Alisher you have still conflicts, please resolve them

**Review by ghost — APPROVED** [review]

File: src/main/resources/application.yml (hunk 8 above)
Comment (AleeWeeR) [reply src/main/resources/application.yml]: ### You can just merge master to this config files by not accepting my configs.

File: src/main/resources/db.changelog/changes/init.yaml (hunk 9 above)
Comment (AleeWeeR) [reply src/main/resources/db.changelog/changes/init.yaml]: I have deleted including init file changes in the master changeset. It won't create a test table.

**Comment by AleeWeeR** [comment]
> 

Done I think we can approve the pr now

**Review by believer-0910 — APPROVED** [review]

**Comment by believer-0910** [comment]
Alisher you have some conflicts that must be resolved
After fixed I will merge with master

**Comment by believer-0910** [comment]
also update your branch, one more branch merged with master

**Review by ghost — APPROVED** [review]

**Pull request closed without merging**